python.asciidoc

```diff
@@ -0,0 +1,562 @@
+regular class but also defines the unit of concurrency. Think of it like
+a class running in its own (lightweight) thread. The main entrypoint of
+a program is always via construction of its `Main` actor.
+
+Pony classes and actors have constructors like Python, they are also
+introduced with the `new` keyword. Regular methods (aka functions) are
+introduced with the `fun` keyword. The body of constructors and
+functions starts after the `⇒`.
+
+Since Pony has no global or static variables, programs need to get their
+process environment passed in as an argument to this initial `Main`
+actor’s constructor. This `Env` object provides access to all the things
+that a process gets from the operating system, such as command line
+arguments, environment variables and the stdio file handles.
+
+This `helloworld` program prints its greeting to stdout in the supplied
+env’s `out` field.
+
+To run the program, execute these commands in the helloworld directory:
+
+[source,shell]
+----
+$ ponyc .
+$ ./helloworld
+----
+
+The `ponyc` command compiles the Pony source in the given directory into
+a native binary which is self-contained and can be run directly. You’ll
+see how small this complete binary is and how fast it starts and runs.
+And, it is self-contained, needing no additional runtime to be
+installed.
+
+[[concurrent-hello-world]]
+Concurrent Hello World
+~~~~~~~~~~~~~~~~~~~~~~
+
+Let's look at what the equivalent parallel example would look like:
+
+[source,python]
+----
+from multiprocessing import Process
+
+
+def greet():
+    print("Hello world!")
+
+
+if __name__ == "__main__":
+    processes = []
+
+    for i in xrange(10):
+        p = Process(target=f)
+        p.start()
+        p.join()
+----
+
+Here’s a rough port to Pony:
+
+[source,ponylang]
+----
+use "collections"
+
+actor Main
+  new create(env: Env) =>
+    for i in Range[U8](1, 10) do
+      Greeter(env.out).greet()
+    end
+
+actor Greeter
+  let _out: StdStream
+
+  new create(out: StdStream) =>
+    _out = out
+
+  be greet() =>
+    _out.print("Hello world!")
+----
+
+That’s it. Note the stuff that’s similar to Python:
+
+* Variables are in `snake_case`
+* We have 'blocks' that use `do` and `end`
+* Variables, while statically typed, have inference, so we don’t need to
+declare types
+
+Here’s some stuff that’s different:
+
+* slightly different syntax, `be` rather than `def` which is an
+asynchronous function which we'll come to later
+
+[[adding-a-little-class]]
+Adding a Little Class
+~~~~~~~~~~~~~~~~~~~~~
+
+In this next example, let’s look at a program with a couple of classes
+that have methods, variables and loops.
+
+Here is the Python version:
+
+[source,python]
+----
+$ cat measurements/rectangle.py
+
+class Rectangle(object):
+    def __init__(self, width, height):
+        self._width = width
+        self._height = height
+
+    def dimensions(self):
+        return "{w} {h}".format(w=self._width, h=self._height)
+
+    def circumference(self):
```

Comment:
Not really a Python/Pony issue, but I'm not sure `circumference` is the right function name here. "Circumference" is most commonly used to refer to perimeter of ellipses (and, by extension, circles), not rectangles. I see that the Java chapter uses the same name, and I'd rather fix it in both places than propagate the error.